x/httpproxy/connect_handler.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpproxy
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type handler struct {
+	dialer transport.StreamDialer
+}
+
+var _ http.Handler = (*handler)(nil)
+
+// ServeHTTP implements [http.Handler].ServeHTTP for CONNECT requests, using the internal [transport.StreamDialer].
+func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// TODO(fortuna): For public services (not local), we need authentication and drain on failures to avoid fingerprinting.
+	if r.Method != http.MethodConnect {
```

Comment:
Another potential TODO item, we may send the request directly to the target server without proxying if method is not supported.